src/storage/store/engine.rs

```diff
@@ -80,42 +74,53 @@ impl Engine for SnapchainEngine {
         true
     }
 
-    fn commit_state_change(&mut self, _shard_state_change: ShardStateChange) {
+    pub fn commit_shard_chunk(&mut self, shard_chunk: ShardChunk) {
         // Create a db transaction
         // Replay the state change
         // If the state root does not match or any of the messages fail to merge, panic?
         // write the events to the db
         // Commit the transaction
         // Emit events
+        match self.shard_store.put_shard_chunk(shard_chunk) {
+            Err(err) => {
+                error!("Unable to write shard chunk to store {}", err)
+            }
+            Ok(()) => {}
+        }
     }
 
-    fn commit_block(&mut self, block: Block) {
-        // Commit the individual shards state changes
-        for shard_chunks in block.shard_chunks.clone() {
-            let shard_index = shard_chunks
-                .header
-                .clone()
-                .unwrap()
-                .height
-                .unwrap()
-                .shard_index;
-            self.commit_state_change(ShardStateChange {
-                shard_id: shard_index,
-                new_state_root: shard_chunks.header.unwrap().shard_root,
-                transactions: shard_chunks.transactions,
-            });
+    pub fn get_confirmed_height(&self) -> Height {
+        match self.shard_store.max_block_number() {
+            Ok(block_num) => Height::new(self.shard_id, block_num),
+            Err(_) => Height::new(self.shard_id, 0),
         }
+    }
+}
+
+pub struct BlockEngine {
+    block_store: BlockStore,
+    shard_id: u32,
+}
+
+impl BlockEngine {
+    pub fn new(shard_id: u32, block_store: BlockStore) -> Self {
```

Comment:
Since we know this is always for the block, we don't need to accept the shard id, can use use a constant of 0.